### PR TITLE
[SW-1193][rel-2.1] Expose 'weightCol' parameter for DL, GBM, GLM, XGBoost (#1132)

### PR DESCRIPTION
**Pull request by mn-mikke**
The PR propagates changes from #1132 to rel-2.1 branch.

### CODE REVIEW

**Comment by mn-mikke** [comment]
@jakubhava There were some confilicts during cherry-picking. Please can you take a look?

**Comment by jakubhava** [comment]
@mn-mikke just fyi: this is the exact part where rel-2.1 differs from other release branches. I want to extract this part as we briefly discussed

File: core/src/main/scala/water/util/DeprecatedMethodMacro.scala [hunk 1]
```diff
@@ -63,7 +63,7 @@ object DeprecatedMethodMacro {
           val warnMessage = s"The method '$methodName' is deprecated. $daMessage"
 
           q"""${Modifiers(f, p, da :: a)} def $methodName[..$tpes](...$args): $returnType =  {
-            water.util.Log.warn($warnMessage)
+            logWarning($warnMessage)
```
Comment (jakubhava) [inline core/src/main/scala/water/util/DeprecatedMethodMacro.scala]: Spark 2.1 does not have the log methods in the internal package

**Comment by mn-mikke** [comment]
@jakubhava Appologize, I merged the branch by accident. See the [fix](https://github.com/h2oai/sparkling-water/commit/29d9194687f6610b121a69f412fe44d1a5560da7) for invisible ```Logging```.

**Pull request merged**